internal/status/cache.go

```diff
@@ -87,3 +106,45 @@ func (c Cache) getProxyStatusUpdates() []k8s.StatusUpdate {
 	return psu
 
 }
+
+func (c Cache) GetProxyValidConditions() map[types.NamespacedName]projcontour.DetailedCondition {
```

Comment:
This is only used by `(*DAG) Statuses()`, which is only used in a test. Can we remove it (and the caller)?